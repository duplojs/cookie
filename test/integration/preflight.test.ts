import { Duplo, useBuilder, Response, zod } from "@duplojs/core";
import { cookieProcess } from "@duplojs/cookie/preflight";
import "@duplojs/node";

describe("cookie preflight", () => {
	const duplo = new Duplo({
		environment: "TEST",
		host: "localhost",
		port: 14092,
	});

	it("should get cookie", async() => {
		const route = useBuilder()
			.preflight(
				cookieProcess,
				{
					options: {},
				},
			)
			.createRoute("GET", "/test-cookie")
			.extract({
				cookies: {
					test: zod.string(),
				},
			})
			.handler(
				(pickup) => {
					const test = pickup("test");

					return new Response(200, "test", test);
				},
			);

		duplo.register(cookieProcess);
		duplo.register(route);

		const server = await duplo.launch();

		const result = await fetch(
			"http://localhost:14092/test-cookie",
			{
				method: "GET",
				headers: {
					Cookie: "test=cookie",
				},
			},
		);

		expect(await result.text()).toMatchSnapshot();

		server.close();
	});
});
