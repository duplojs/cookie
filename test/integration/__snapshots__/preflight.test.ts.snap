// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cookie preflight > should get cookie 1`] = `"cookie"`;
